ibis/backends/sql/compilers/base.py

```diff
@@ -671,12 +671,28 @@ def fn(node, __unused__, **kwargs):
         elif isinstance(out, sge.Subquery):
             out = out.this
 
-        # add cte definitions to the select statement
+        merged_ctes = []
         for cte in ctes:
-            alias = sg.to_identifier(aliases[cte], quoted=self.quoted)
-            out = out.with_(
-                alias, as_=results[cte].this, dialect=self.dialect, copy=False
+            this = results[cte]
+            if "alias" in results[cte].args:
+                this = this.this
+            modified_cte = sg.exp.CTE(
+                alias=sg.to_identifier(aliases[cte], quoted=self.quoted), this=this
             )
+            merged_ctes.append(modified_cte)
+        merged_ctes = [*merged_ctes, *out.ctes]
```

Comment:
```suggestion
        merged_ctes.extend(out.ctes)
```

No need to create a new list, you can modify `merged_ctes`.

Reply:
Sometimes I get too functional-programming-happy. Fixed!